### PR TITLE
Feature/1710

### PR DESCRIPTION
**Pull request by kjugi**
### Related issues

#1710

### Short description and why it's useful

Allows user to choose one category from all searched products from entered word and filter only this which contains it

### Screenshots of visual changes before/after (if there are any)

zrzut ekranu 2019-01-18 o 18 37 47

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [ ] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [ ] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

File: src/themes/default/components/core/blocks/Category/CategoryPanel.vue [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    }
+  },
+  methods: {
+    filterProducts (category, buttonIndex) {
+      this.isFilterHidden = buttonIndex
+      this.$store.dispatch('category/addSidebarSelectedCategory', category)
+    },
+    deleteFilter () {
+      this.isFilterHidden = false
+      this.$store.dispatch('category/deleteSidebarSelectedCategory')
```
Comment (patzick) [inline src/themes/default/components/core/blocks/Category/CategoryPanel.vue]: same here, emit with no with null
`this.$emit('input', null)`

File: src/themes/default/components/core/blocks/SearchPanel/SearchPanel.vue [hunk 2]
```diff
@@ -65,11 +73,46 @@
...
+    ProductTile,
+    CategoryPanel
+  },
+  computed: {
+    categories () {
+      return this.products.map(product => {
+        return product.category
+      })
+    },
+    selectedCategory () {
```
Comment (patzick) [inline src/themes/default/components/core/blocks/SearchPanel/SearchPanel.vue]: this should be in this component data section binded to category-panel by v-model

File: src/themes/default/components/core/blocks/SearchPanel/SearchPanel.vue [hunk 3]
```diff
@@ -38,15 +38,23 @@
           </div>
         </div>
       </div>
+      <div v-if="allProducts.length > 0 && categories.length > 1" class="categories">
+        <category-panel :product-categories="categories" />
```
Comment (patzick) [inline src/themes/default/components/core/blocks/SearchPanel/SearchPanel.vue]: here we could add `v-model="selectedCategory"`

File: src/themes/default/components/core/blocks/SearchPanel/SearchPanel.vue [hunk 4]
```diff
@@ -65,11 +73,46 @@
...
+        return product.category
+      })
+    },
+    selectedCategory () {
+      return this.$store.state.category.sidebar_selected_category
+    }
+  },
+  data () {
+    return {
+      allProducts: this.products || []
```
Comment (patzick) [inline src/themes/default/components/core/blocks/SearchPanel/SearchPanel.vue]: it's better to change this to computed property like `visibleProducts`, which returns this.products filtered by this.selectedCategory if is set. Then we don't these two watchers below

**Comment by patzick** [comment]
Hey @kjugi, are you willing to finish this PR?:)

**Comment by pkarw** [comment]
@patzick I belive You should take care of the required changes during this 1h / daily bugs/Slack duties we’ve discussed OK? Would be great to finish this task till end of the next week

**Comment by patzick** [comment]
Due to no response from author I'm closing this PR in favour of #2355.
Thanks @kjugi for your input!

**Pull request closed without merging**